dimod/io/coo.py

```diff
@@ -0,0 +1,76 @@
+import re
+
+from dimod.binary_quadratic_model import BinaryQuadraticModel
+
+_LINE_REGEX = r'^\s*(\d+)\s+(\d+)\s+([+-]?([0-9]*[.])?[0-9]+)\s*$'
```

Comment:
Ok, so this addresses most of my previous comments. We could also constrain floats to non-zero first digit (for multiple digits), but that would probably just overly-complicate this simple regex, and `float()` will happily parse those anyway. Since COO is not really a specified format, we can be generous on what we parse.

Btw, you can also slightly optimize the parsing by compiling this regex in module scope, not function calls.

Reply:
Btw2, you can simplify regex by using `\d` in place of `[0-9]`.